libs/boost/Makefile

```diff
@@ -320,6 +347,10 @@ TARGET_CFLAGS += \
 	$(if $(CONFIG_PACKAGE_boost-python3), -I$(STAGING_DIR)/usr/include/python3.5/) \
 	$(if $(CONFIG_SOFT_FLOAT),-DBOOST_NO_FENV_H) -fPIC
 
+TARGET_CXXFLAGS += \
+	$(if $(or $(CONFIG_GCC_USE_VERSION_5), $(CONFIG_GCC_USE_VERSION_6)), -std=c++14, -std=c++11)
```

Comment:
There's a bug here. Please hold the merge of this commit.

It is not intended to be like this. It should use C++11 on GCC_VERSION_4_8 detection or C++14 otherwise.